drop-transfer/src/ws/events.rs

```diff
@@ -126,21 +162,24 @@ impl<T: Transfer> FileEventTx<T> {
         let mut lock = self.inner.write().await;
 
         if let Some(started) = lock.started.take() {
-            let info = self.file_info();
-
-            lock.moose.service_quality_transfer_file(
-                Err(status as _),
-                self.xfer.id().to_string(),
-                started.elapsed().as_millis() as _,
-                T::direction(),
-                info,
-            );
+            let file_info = self.file_info();
+
+            lock.moose.event_transfer_file(TransferFileEventData {
+                phase: drop_analytics::TransferFilePhase::Finished,
+                transfer_id: self.xfer.id().to_string(),
+                transfer_time: started.elapsed().as_millis() as i32,
+                path_id: file_info.path_id,
+                direction: file_info.direction,
+                transferred: utils::to_kb(lock.transferred),
+                result: status as _,
+            });
         }
     }
 }
 
 impl FileEventTx<IncomingTransfer> {
     pub async fn progress(&self, transfered: u64) {
+        self.inner.write().await.transferred = transfered;
```

Comment:
I don't this is correct. The `emit()` function locks on its own, so I suggest adjusting this function so that the locking takes place only once here.